internal/service/image.go

```diff
@@ -18,8 +18,13 @@ func (h *ServiceHandler) GetImage(ctx context.Context, request server.GetImageRe
 	if !ok {
 		return server.GetImage500JSONResponse{Message: "error creating the HTTP stream"}, nil
 	}
-	ova := &image.Ova{SshKey: request.Params.SshKey, Writer: writer}
 
+	source, err := h.store.Source().Get(ctx, request.Params.Id)
+	if err != nil {
+		return server.GetImage404JSONResponse{}, nil
+	}
+
+	ova := &image.Ova{SourceID: source.ID, AgentID: source.Agent.ID, SshKey: request.Params.SshKey, Writer: writer}
```

Comment:
Why we need to pass agentid?

Reply:
because of the first request from agent to update it's status.  You need the id now.
1st req -> update agent status
2st req -> update source

Reply:
Wouldn't it be better to have just single request as before?

Reply:
IMO no. Better in what way? because the agent makes only one req? 
These are two resources with different lifecycles. Keeping them separately allows us to change easily the behavior in the future. Take for example the event streamer. I want to know when the agent changed state and when the inventory has been updated (with timestamp and content). It will be very hard if you have everything in one req.
Moreover, it will be a mess in the service layer trying to update different tables from one shot and I really really don't want to rollback to previous model.

Reply:
In that case you still don't need the AgentID in OVA template, no? You can generate it as it was in multisource, no?

Reply:
If the source don't have any agent associated, we won't display anything. We don't wait for credentials, we don't have any agent yet, so there is nothing to wait for, it may even be disconnected env, so we will never get any creds IP. When the agent boots and registers at that point we wait for credentials, and at that point we associate it with source.